packages/flutter/test/material/carousel_test.dart

```diff
@@ -1413,6 +1413,63 @@ void main() {
       expect(buttonPressed, isTrue);
     },
   );
+
+  // Regression test for https://github.com/flutter/flutter/issues/160679
+  testWidgets('CarouselView does not crash if itemExtent is zero', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: SizedBox(
+            width: 100,
+            child: CarouselView(
+              itemExtent: 0,
+              children: <Widget>[Container(color: Colors.red, width: 100, height: 100)],
+            ),
+          ),
+        ),
+      ),
+    );
+
+    expect(tester.takeException(), isNull);
+  });
+
+  // Regression test for https://github.com/flutter/flutter/issues/160679
+  testWidgets('CarouselView does not crash if layout constraints are zero', (
```

Comment:
Can we add a layout pass where there is room to layout?  As in, one with width and height set >0 to verify it lays out, can go to zero, and then return to that state?

Reply:
Will try to add a test for that!